sql-server-2014-alwayson-existing-vnet-and-ad/azuredeploy.json

```diff
@@ -2,14 +2,12 @@
   "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
   "contentVersion": "1.0.0.0",
   "parameters": {
-
-    "newStorageAccountNamePrefix": {
+    "namePrefix": {
```

Comment:
I applied the updates recommended above.  Note that I included uniqueString in the storage account naming, but am also including a name prefix in front of it - just in case someone deploys this template into an existing resource group where storage accounts have already been named with uniqueString (to avoid conflicts).  In terms of DSC agent extension, prior versions of this agent extension have introduced breaking changes with how DSC configs are applied in the VM (which broke templates), but there doesn't appear to be any issues with version 2.17 and later, so I've set autoupdate back to true.  I've tested all changes and they successfully deploy.

Let me know if there's anything else I should address to permit this PR to be merged.  Thanks!